### DISCUSSION TITLE
Problems running the Script in PS 2023

### DISCUSSION THREAD

**Discussion by angusin** [joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#22] [General]
Hello! Thanks for creating this script, I didn't manage to run it yet, but hopefully I will solve it.

My problem is that when I run the script, the PSD opens and flashes for a second and then the success alert appears:

![image](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/assets/8282153/4298db14-68cb-4eb2-95c7-eb223af30ca8)

The output folder is created but nothing is inside.

My machine:
**OS: Mac OS 13.3.1
PS: 2023 24.0.0**

My config is:

![image](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/assets/8282153/f8a2fe47-894a-4194-9fad-60cb0a09e3ce)

My folder structure:

![image](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/assets/8282153/ce7915da-c3c8-48fb-9fdd-dd78d1f53236)

And the target layer:

![image](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/assets/8282153/0c03f2f1-bf71-4608-91b5-17e5abc5e300)

Do you have any clue what might be happening?

Thanks again for taking the time to create this script!

**Comment by joonaspaakko** [comment]
Can't see anything wrong with it... The only thing I can think of is that maybe you didn't save the mockup psd after naming the layer (probably not), but otherwise, it all looks fine.

Out of curiosity, I took the [standalone example](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement.js-Photoshop-script-standalone-example) and [tried to modify it to match your setup](https://www.dropbox.com/sh/fgze7s0jfq177y2/AABFRYuy3CkLarv4mkddgGcCa?dl=0) but I could not replicate the issue. 

What you might want to do is download the standalone example or my attempt at copying your exact situation and first of all see if it works without changing anything and then add in your own files and see if it works that way.

**Reply by angusin** [reply]
Thanks @joonaspaakko for the prompt reply. Sadly, no luck either way. I even create a really simple file with only the smart object and nothing; the script launches, it opens for a milisecond the mockup.psd file, it closes it, and the success message appears.

I have tried many things but no luck. Maybe it's because some conflict between my version of PS and my Mac... idnk

**Reply by joonaspaakko** [reply]
Did you download either starting point and try them without changing anything? If they don't work like that, there's probably some kind of issue with your PS version. If they do work without you changing anything, there's likely some kinda issue in your settings script or mockup file that isn't super obvious.

The script has been tested in PS CC and most recently in PS CC 2023, which I believe you have. It's also been very lightly tested in PS CS6, though I can't guarantee that works perfectly. Anything older than that and I'm pretty sure it won't work.

**Reply by spaceda1** [reply]
I'm running latest script on v24.1.1 without any problems. 
check paths as script stops silently if eg psd is not found

**Reply by angusin** [reply]
Thanks @spaceda1 , I'm using 24.0.0.
Paths are good, the mockup is opening for a mere second and then closes, so the path is good (images on the original post). Also, the script is actually ending in the same second the mockup is closed, with a success message, and telling that everything is good, but it's not.
As the example is also not working for me, I'm going to try in a different computer, because 100% sure it's something about this one.
Thanks! I'll let you know

**Reply by angusin** [reply]
I'm back and with good news!
In a Windows computer with PS 2023 (24.x.x), everything works fine. So it's solved and for sure it was something with my computer (MPB M1 2019 with PS 2023). It was not about the smart object or the PSD mockup.
Only thing missing: thanks again @joonaspaakko for an inmense work, you have save me many hours of work. 🥇

**Reply by joonaspaakko** [reply]
Great 👍 If you want to debug what is going on on that other computer... It's not particularly easy, but possible.
